functions/Start-DbaAgentJob.ps1

```diff
@@ -187,7 +191,13 @@ function Start-DbaAgentJob {
                 # Start the job
                 $lastrun = $currentjob.LastRunDate
                 Write-Message -Level Verbose -Message "Last run date was $lastrun"
-                $null = $currentjob.Start()
+                if ($StepName) {
+                    Write-Message -Level Verbose -Message "Starting job [$currentjob] at step [$StepName]"
+                    $null = $currentjob.Start($StepName)
```

Comment:
Do we need to validate the step name exists?

Reply:
At the moment if the step name does not exist the job will not run.
How do we want to handle this?
Warning and stop?
Error and stop?

What's the standard in the module?

Reply:
Will not run how? Errors with a clear message, errors with red text?

Reply:
At the moment it will silently not run with no errors.

Reply:
Is that the type of output you would expect?

Reply:
It's either that and/or a warning/error saying the step does not exist. 

Does not matter to me. What would be standard in dbatools?

Reply:
as long as it is consistent with itself ..... what happens if you try to start a job that doesn't exist ?

Reply:
> as long as it is consistent with itself ..... what happens if you try to start a job that doesn't exist ?

Good point. Will check it as I don't know from the top of my head.

Reply:
For the Job parameter, the command does not return anything if the job does not exist. 
For the StepName parameter 
- added check for the step (if exists in job's steps)
- added **continue** as without it the command goes into a loop and with **return** it will stop at the first job that does not have the specified StepName;